WalletWasabi/WabiSabi/Backend/CoinJoinMempoolManager.cs

```diff
@@ -0,0 +1,53 @@
+using NBitcoin;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using WalletWasabi.Bases;
+using WalletWasabi.WabiSabi.Backend.Rounds.CoinJoinStorage;
+using System.Collections.Immutable;
+using WalletWasabi.BitcoinCore.Mempool;
+
+namespace WalletWasabi.WabiSabi.Backend;
+
+public class CoinJoinMempoolManager : IDisposable
+{
+	private bool _disposedValue;
+
+	public CoinJoinMempoolManager(ICoinJoinIdStore coinJoinIdStore, IMempoolMirror mempoolMirror)
+	{
+		CoinJoinIdStore = coinJoinIdStore;
+		MempoolMirror = mempoolMirror;
+		MempoolMirror.Tick += Mempool_Tick;
+	}
+
+	private ICoinJoinIdStore CoinJoinIdStore { get; }
+	private IMempoolMirror MempoolMirror { get; set; }
+	public ImmutableArray<uint256> CoinJoinIds { get; private set; } = ImmutableArray.Create<uint256>();
+
+	private void Mempool_Tick(object? sender, TimeSpan e)
+	{
+		var mempoolHashes = MempoolMirror?.GetMempoolHashes() ?? throw new InvalidOperationException("Mempool provider missing.");
```

Comment:
`MempoolMirror`  was nullable because of a problem in the design. Now it is not nullable any more. 
```suggestion
		var mempoolHashes = MempoolMirror.GetMempoolHashes();
```

Reply:
Fixed!